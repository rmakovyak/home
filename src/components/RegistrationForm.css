.RegistrationForm {
  padding: 20px;
}
