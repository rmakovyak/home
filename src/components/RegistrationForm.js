import React, { Component } from "react";

import ProgressBar from "./ProgressBar";
import styles from "./RegistrationForm.css";

class RegistrationForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      level: 0
    };
  }

  render() {
    return (
      <div className="RegistrationForm">
        <ProgressBar level={this.state.level} />
        <button onClick={() => this.setState({ level: this.state.level + 25 })}>
          Progress
        </button>
      </div>
    );
  }
}

export default RegistrationForm;
