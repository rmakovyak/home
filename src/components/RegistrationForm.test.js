import React from 'react';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import RegistrationForm from './RegistrationForm';

configure({ adapter: new Adapter() });

test('render a RegistrationForm', () => {
  const wrapper = shallow(<RegistrationForm />);
  expect(wrapper).toMatchSnapshot();
});
