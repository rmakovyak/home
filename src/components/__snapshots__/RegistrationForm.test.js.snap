// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render a RegistrationForm 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RegistrationForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ProgressBar
          level={0}
/>,
        <div
          className="RegistrationForm__group"
>
          <label>
                    Name
          </label>
          <input
                    name="name"
                    onChange={[Function]}
                    type="text"
                    value=""
          />
          <p
                    className="RegistrationForm__error"
          />
</div>,
        false,
        false,
        false,
        false,
        <div
          className="RegistrationForm__control"
>
          <button
                    onClick={[Function]}
                    type="submit"
          >
                    Next
          </button>
</div>,
      ],
      "className": "RegistrationForm",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "level": 0,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label>
              Name
</label>,
            <input
              name="name"
              onChange={[Function]}
              type="text"
              value=""
/>,
            <p
              className="RegistrationForm__error"
/>,
          ],
          "className": "RegistrationForm__group",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Name",
            },
            "ref": null,
            "rendered": "Name",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "name",
              "onChange": [Function],
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "RegistrationForm__error",
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
        ],
        "type": "div",
      },
      false,
      false,
      false,
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            false,
            <button
              onClick={[Function]}
              type="submit"
>
              Next
</button>,
          ],
          "className": "RegistrationForm__control",
        },
        "ref": null,
        "rendered": Array [
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Next",
              "onClick": [Function],
              "type": "submit",
            },
            "ref": null,
            "rendered": "Next",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ProgressBar
            level={0}
/>,
          <div
            className="RegistrationForm__group"
>
            <label>
                        Name
            </label>
            <input
                        name="name"
                        onChange={[Function]}
                        type="text"
                        value=""
            />
            <p
                        className="RegistrationForm__error"
            />
</div>,
          false,
          false,
          false,
          false,
          <div
            className="RegistrationForm__control"
>
            <button
                        onClick={[Function]}
                        type="submit"
            >
                        Next
            </button>
</div>,
        ],
        "className": "RegistrationForm",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "level": 0,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                Name
</label>,
              <input
                name="name"
                onChange={[Function]}
                type="text"
                value=""
/>,
              <p
                className="RegistrationForm__error"
/>,
            ],
            "className": "RegistrationForm__group",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Name",
              },
              "ref": null,
              "rendered": "Name",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "name",
                "onChange": [Function],
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "RegistrationForm__error",
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
          ],
          "type": "div",
        },
        false,
        false,
        false,
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              false,
              <button
                onClick={[Function]}
                type="submit"
>
                Next
</button>,
            ],
            "className": "RegistrationForm__control",
          },
          "ref": null,
          "rendered": Array [
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Next",
                "onClick": [Function],
                "type": "submit",
              },
              "ref": null,
              "rendered": "Next",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
